src/panels/alarm-configuration/AlarmConfigurationPanel.tsx

```diff
@@ -0,0 +1,158 @@
+    [data.request]
+  );
+
+  useEffect(() => {
+    if (results.invalidFormat || queryInfo.invalidFormat) {
+      setWarnings(results.warning + ' ' + queryInfo.warning);
+    } else {
+      setWarnings('');
+    }
+
+    const templateSrv = getTemplateSrv();
+    setAlarmName(queryInfo.alarmName ? templateSrv.replace(queryInfo.alarmName) : '');
+    setEntityId(queryInfo.entityId ? templateSrv.replace(queryInfo.entityId) : '');
+    setAlarmThreshold(results.alarmThreshold);
+    setAlarmNotificationRecipient(results.alarmNotificationRecipient ? results.alarmNotificationRecipient : '');
+  }, [results, queryInfo]);
+
+  const updateAlarmThreshold = useCallback(
+    (newThreshold: number) => {
+      const request: BatchPutPropertyValuesRequest = {
+        workspaceId: dataSourceParams?.workspaceId!,
+        entries: [
+          {
+            entityPropertyReference: {
+              componentName: alarmName,
+              entityId: entityId,
+              propertyName: 'alarm_threshold',
+            },
+            propertyValues: [
+              {
+                timestamp: new Date(),
+                value: {
+                  doubleValue: newThreshold,
+                },
+              },
+            ],
+          },
+        ],
+      };
+      if (twinMakerClient) {
+        twinMakerClient
+          .batchPutPropertyValues(request)
+          .promise()
+          .then(() => {
+            setAlarmThreshold(newThreshold);
+          })
+          .catch((error: any) => {
+            setWarnings(error.message);
+          });
+      } else {
+        console.error('TwinMaker client not defined');
+      }
+    },
+    [dataSourceParams, alarmName, entityId, twinMakerClient, setAlarmThreshold, setWarnings]
+  );
+
+  const [editModalOpen, setEditModalOpen] = useState(false);
+  const toggleModal = useCallback(() => {
+    setEditModalOpen(!editModalOpen);
+  }, [setEditModalOpen, editModalOpen]);
+
+  if(!configured) {
+    return <LoadingPlaceholder text={'Loading...'} />;
+  } else if (!dataSourceParams) {
+    return <div> No TwinMaker Data Source Connected: ${options.datasource} </div>;
+  } else if (warnings) {
+    return (
+      <div>
+        <div> TwinMaker Data Source Connected </div>
+        <div> Warnings: {warnings}</div>
+      </div>
+    );
+  } else {
+    return (
+      <div>
+        <Label description={alarmName}> Alarm ID </Label>
+        <br />
+        <Label description={alarmThreshold}> Thresold </Label>
+        <br />
+        <Label description={alarmNotificationRecipient}> Notifications</Label>
+        <div>
+          <Button variant="secondary" onClick={toggleModal}>Edit Alarm</Button>
+        </div>
+        <AlarmEditModal
+          isOpen={editModalOpen}
+          onDismiss={toggleModal}
+          onSave={updateAlarmThreshold}
+          currentThreshold={alarmThreshold!}
+        ></AlarmEditModal>
+      </div>
```

Comment:
This should be a separate component. Typically you want entry points like this to be basic layout components:

```
<Layout>
  <StateProvider>
       <Panel />
  </StateProvider>
  </Layout>
  ```
  
  Basically, the entry point component only responsible for registring and destroying when it's loaded in grafana. Everything else gets handed off to other things. Anything grafana gives you on initilaization should get passed to the state provider via props, and then handled internally in there.
  
 Child components, then can interact with the state via context, and are responsible for their own little chunk of the view.